nms/app/views/equipment/GatewayDetailConfigEdit.tsx

```diff
@@ -1303,6 +1324,147 @@ export function HeaderEnrichmentConfig(props: Props) {
           )}
         </List>
       </DialogContent>
+      <DialogActions>
+        <Button onClick={props.onClose}>Cancel</Button>
+        <Button
+          onClick={() => void onSave()}
+          variant="contained"
+          color="primary">
+          {props.isAdd ? 'Save And Continue' : 'Save'}
+        </Button>
+      </DialogActions>
+    </>
+  );
+}
+
+export function NextGenerationCoreConfig(props: Props) {
+  const enqueueSnackbar = useEnqueueSnackbar();
+  const [error, setError] = useState('');
+  const ctx = useContext(GatewayContext);
+
+  const [ngcConfig, setNgcConfig] = useState<GatewayNgcConfigs>(
+    props.gateway.cellular.ngc ?? {},
+  );
+
+  useEffect(() => {
+    setNgcConfig(props.gateway.cellular.ngc ?? {});
+    setError('');
```

Comment:
Why do we need the `setError('')` here?

Reply:
It's updating the dialog state in case the underlying config state changes. I'm not sure of all cases when this might happen (it was basically copied from the other edit dialogs), but since any printed errors are likely related to the old state, it makes sense to reset the error message as well.